lib/Mount/FolderStorageManager.php

```diff
@@ -54,7 +57,72 @@ public function getRootAndStorageIdForFolder(int $folderId): array {
 	/**
 	 * @param 'files'|'trash'|'versions' $type
 	 */
-	public function getBaseStorageForFolder(int $folderId, ?FolderDefinition $folder = null, ?IUser $user = null, bool $inShare = false, string $type = 'files'): IStorage {
+	public function getBaseStorageForFolder(
+		int $folderId,
+		bool $separateStorage,
+		?FolderDefinition $folder = null,
+		?IUser $user = null,
+		bool $inShare = false,
+		string $type = 'files',
+	): IStorage {
+		if ($separateStorage) {
+			return $this->getBaseStorageForFolderSeparateStorageLocal($folderId, $folder, $user, $inShare, $type);
+		} else {
+			return $this->getBaseStorageForFolderRootJail($folderId, $folder, $user, $inShare, $type);
+		}
+	}
+
+	/**
+	 * @param 'files'|'trash'|'versions' $type
+	 */
+	public function getBaseStorageForFolderSeparateStorageLocal(
+		int $folderId,
+		?FolderDefinition $folder = null,
+		?IUser $user = null,
+		bool $inShare = false,
+		string $type = 'files',
+	): IStorage {
+		$dataDirectory = $this->config->getSystemValue('datadirectory');
+		$rootPath = $dataDirectory . '/__groupfolders/' . $folderId;
+		$init = !is_dir($rootPath);
+		if ($init) {
+			mkdir($rootPath, 0777, true);
```

Comment:
777 permission doesn't seem like a good idea? Should be 750 or something like that

Reply:
`0777` is the default argument, the folder won't actually be `0777` because of the umask though